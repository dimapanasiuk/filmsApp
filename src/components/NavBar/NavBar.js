import React from 'react';
import { arrayOf, string, func, object } from 'prop-types';
import { v4 as uuid4 } from 'uuid';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { Button } from '@material-ui/core';

import { chooseCategory } from '../../redux/categoryData/categoryDataActions';

class NavBar extends React.Component {
  clickHandlerCategoryChoose = (e, foo) => {
    let category = e.target.innerText;
    this.props.dispatch(chooseCategory(category));
    foo();
  };

  htmlData = (arr, foo) => {
    return arr.map((i) => {
      return (
        <Button
          className="navbar-button"
          variant="contained"
          color="primary"
          key={uuid4()}
          onClick={(e) => this.clickHandlerCategoryChoose(e, foo)}
          to={`/categories/${i}`}
          component={Link}
        >
          {i}
        </Button>
      );
    });
  };

  render() {
    let { list, resetPagination } = this.props;

    return (
      <div className="categories-content">
        {this.htmlData(list, resetPagination)}
      </div>
    );
  }
}

NavBar.propTypes = {
  list: arrayOf(string),
  dispatch: func,
  chooseCategory: object,
  resetPagination: func,
};

const mapStateToProps = (state) => {
  return state;
};

export default connect(mapStateToProps)(NavBar);
