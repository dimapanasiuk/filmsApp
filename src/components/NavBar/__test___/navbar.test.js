import React from "react";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from 'react-router-dom';

import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import NavBar from "../NavBar";

import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});


const middlewares = [];

const mockStore = configureStore(middlewares);
const actionFire = () => ({ type: 'CHOOSECATEGORY', data: 'test' });


it("renders with or without a opportunity", () => {
  const initialState = {};
  const store = mockStore(initialState);
  store.dispatch(actionFire());

  const actions = store.getActions();
  const expectedPayload = { type: 'CHOOSECATEGORY', data: 'test' };

  expect(actions).toEqual([expectedPayload]);

  act(() => {
    render(
      <Provider store={store}>
        <NavBar list={[]} />
      </Provider>, container);
  });

  expect(container.textContent).toBe("");
  expect(document.querySelector('ul')).toBeEmptyDOMElement();


  act(() => {
    render(
      <Provider store={store}>
        <Router>
          <NavBar list={['test', 'data']} />
        </Router>
      </Provider>, container);
  });

  const listValues = container.getElementsByTagName("li");

  expect(listValues[0].textContent).toHaveLength(4);
  expect(listValues[1].textContent).toBe("data");
  expect(document.querySelector('li')).toBeVisible();


  const links = container.getElementsByTagName("a");

  links[0].focus();
  expect(links[0]).toHaveFocus();


  act(() => {
    listValues[0].dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });
  expect(listValues[0].textContent).toBe("test");



  const tree = renderer.create(<Provider store={store}>
    <Router>
      <NavBar list={['test', 'data']} />
    </Router>
  </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
