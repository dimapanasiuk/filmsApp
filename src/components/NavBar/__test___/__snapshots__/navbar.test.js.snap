// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with or without a opportunity 1`] = `
<aside>
  <nav>
    <ul>
      <li
        className="category-item"
      >
        <a
          href="/categories/test"
          onClick={[Function]}
        >
          test
        </a>
      </li>
      <li
        className="category-item"
      >
        <a
          href="/categories/data"
          onClick={[Function]}
        >
          data
        </a>
      </li>
    </ul>
  </nav>
</aside>
`;
